lib/lib.go

```diff
@@ -0,0 +1,15 @@
+package lib
+
+import (
+	s "dag/structure"
+)
+
+func CreateNode(root *s.Root, value interface{}) *s.StructNode {
```

Comment:
Instead of taking `root` as parameter, I would return a simple Node and them implement a method `AttachNode` to `Root` so you can easily attach it.

This way you apply single responsibility principle

Reply:
implement a method to root that attach the created node to root and return it's address ?